lib/base-compiler.js

```diff
@@ -112,6 +115,40 @@ export class BaseCompiler {
         this.packager = new Packager();
     }
 
+    getCmakeBaseEnv() {
```

Comment:
Would it be ok to make this async and not use the `sync` calls below? statting a whole bunch of network files in a row seems like we should try not to block the main thread? Probably ok, but this stuff is death by a thousand cuts...

Reply:
trying to do so, but fs.stat is the worst...

Reply:
Ah! Ok. don't fret too much. There's an asyncified version I thought? But if not, disregard my grumpy comments and let's worry about more important things :)

Reply:
yes, but it can throw if you await